app/src/main/java/mod/RockAriful/AndroXStudio/GithubConfigActivity.java

```
@@ -0,0 +1,474 @@
+					}
+					else {
+						if (setRefSpecs.getText().toString().trim().equals("")) {
+							SketchwareUtil.toastError("Please enter RefSpecs/ Branch Reference");
+							enable.setChecked(false);
+						}
+						else {
+							if (username.getText().toString().trim().equals("")) {
+								SketchwareUtil.toastError("Please enter Github username");
+								enable.setChecked(false);
+							}
+							else {
+								if (pass.getText().toString().trim().equals("")) {
+									SketchwareUtil.toastError("Please enter Github access token");
+									enable.setChecked(false);
+								}
+								else {
+                           	    push_data.setVisibility(View.VISIBLE);
+		 	     			     create_token.setVisibility(View.GONE);
+							       url.setEnabled(false);
+						      	 setRefSpecs.setEnabled(false);
+						      	 username.setEnabled(false);
+							       pass.setEnabled(false);
+                    
+									map = new HashMap<>();
+									map.put("repository", url.getText().toString());
+									map.put("RefSpecs", setRefSpecs.getText().toString());
+									map.put("username", username.getText().toString());
+									map.put("token", pass.getText().toString());
+									JsonMAP.add(map);
+									FileUtil.writeFile(FileUtil.getExternalStorageDir()+"/.sketchware/data/"+sc_id+"/github_config", new Gson().toJson(JsonMAP));
+                                    map.clear();
+									JsonMAP.clear();
+								}
+							}
+						}
+					}
+                    
+				}
+				else {
+					FileUtil.writeFile(FileUtil.getExternalStorageDir()+"/.sketchware/data/"+sc_id+"/github_config", "[]");
+                    push_data.setVisibility(View.GONE);
+			    	create_token.setVisibility(View.VISIBLE);
+					url.setEnabled(true);
+			   	 setRefSpecs.setEnabled(true);
+					username.setEnabled(true);
+					pass.setEnabled(true);
+                    try{
+                     map.clear();
+					 JsonMAP.clear();
+                    } catch(Exception e){}

```

Silent error catching shouldn't be done